state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -1810,6 +1895,39 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 			return Ok(())
 		}
 
+		let channel_owner = deposit_channel_details.owner.clone();
+
+		if TaintedTransactions::<T, I>::take(channel_owner.clone(), deposit_details.clone())
+			.is_some()
+		{
+			let refund_address = match deposit_channel_details.action {
+				ChannelAction::Swap { refund_params, .. } =>
+					refund_params.as_ref().map(|refund_params| refund_params.refund_address.clone()),
+				ChannelAction::CcmTransfer { refund_params, .. } =>
+					refund_params.as_ref().map(|refund_params| refund_params.refund_address.clone()),
+				ChannelAction::LiquidityProvision { refund_address, .. } => Some(refund_address),
+			};
+
+			let tainted_transaction_details = TaintedTransactionDetails {
+				refund_address,
+				amount: deposit_amount,
+				asset,
+				tx_id: deposit_details.clone(),
+			};
+
+			ScheduledTxForReject::<T, I>::append(tainted_transaction_details);
```

Comment:
How will this work if the deposit has already been boosted? Right now it looks like you would be able to drain the boost pool if your deposit is rejected.

Reply:
@Janislav ☝️ 

We forgot to fully consider boost in this.

Reply:
Mhhh... I would not call it "forgot" as long it is not live 😅.